src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -0,0 +1,205 @@
+            storage_account = StorageResourceIdentifier(cmd.cli_ctx.cloud, os_disk_uri)
+            # Validate create vm create command to validate parameters before runnning copy disk commands
+            validate_create_vm_command = create_rescue_vm_command + ' --validate'
+            logger.info('Validating VM template before continuing:')
+            _call_az_command(validate_create_vm_command)
+
+            # get storage account connection string
+            get_connection_string_command = 'az storage account show-connection-string -g {g} -n {n} --query connectionString -o tsv' \
+                                            .format(g=resource_group_name, n=storage_account.account_name)
+            logger.info('Fetching storage account connection string:')
+            connection_string = _call_az_command(get_connection_string_command).strip('\n')
+
+            # Create Snapshot of Unmanaged Disk
+            make_snapshot_command = 'az storage blob snapshot -c {c} -n {n} --connection-string "{con_string}" --query snapshot -o tsv' \
+                                    .format(c=storage_account.container, n=storage_account.blob, con_string=connection_string)
+            logger.info('Creating snapshot of OS disk:')
+            snapshot_timestamp = _call_az_command(make_snapshot_command).strip('\n')
+            snapshot_uri = os_disk_uri + '?snapshot={timestamp}'.format(timestamp=snapshot_timestamp)
+
+            # Copy Snapshot into unmanaged Disk
+            copy_snapshot_command = 'az storage blob copy start -c {c} -b {name} --source-uri {source} --connection-string "{con_string}"' \
+                                    .format(c=storage_account.container, name=copied_os_disk_name, source=snapshot_uri, con_string=connection_string)
+            logger.info('Creating a copy disk from the snapshot:')
+            _call_az_command(copy_snapshot_command)
+             # Generate the copied disk uri
+            copied_disk_uri = os_disk_uri.rstrip(storage_account.blob) + copied_os_disk_name
+
+            # Create new rescue VM with copied ummanaged disk command
+            create_rescue_vm_command = create_rescue_vm_command + ' --use-unmanaged-disk'
+            logger.info('Creating rescue vm while disk copy is in progress:')
+            _call_az_command(create_rescue_vm_command)
+
+            logger.info('Checking if disk copy is done:')
+            copy_check_command = 'az storage blob show -c {c} -n {name} --connection-string "{con_string}" --query properties.copy.status -o tsv' \
+                                 .format(c=storage_account.container, name=copied_os_disk_name, con_string=connection_string)
+            copy_result = _call_az_command(copy_check_command).strip('\n')
+            if copy_result != 'success':
+                # TODO, what then if unsucessful. Call again or exit?
+                # Let it run for now and see how the disk state it.
+                logger.warning('Disk copy UNSUCCESSFUL.')
```

Comment:
if it is unsuccessful, will it prompt to clean up resources that already created?

Reply:
This is an edge case that never happened so not sure what to do here. I'll talk with the team on where to go with this.